extension/src/experiments/commands/register.ts

```diff
@@ -153,19 +156,55 @@ const registerExperimentInputCommands = (
     RegisteredCliCommands.EXPERIMENT_BRANCH,
     () =>
       experiments.getCwdExpNameAndInputThenRun(
-        AvailableCommands.EXPERIMENT_BRANCH,
-        Title.ENTER_BRANCH_NAME
+        Title.ENTER_BRANCH_NAME,
+        (cwd, ...args: Args) =>
+          experiments.runCommand(
+            AvailableCommands.EXPERIMENT_BRANCH,
+            cwd,
+            ...args
+          )
       )
   )
 
   internalCommands.registerExternalCliCommand(
     RegisteredCliCommands.EXPERIMENT_VIEW_BRANCH,
     ({ dvcRoot, id }: ExperimentDetails) =>
       experiments.getExpNameAndInputThenRun(
-        AvailableCommands.EXPERIMENT_BRANCH,
         Title.ENTER_BRANCH_NAME,
         dvcRoot,
-        id
+        id,
+        (name: string, input: string) =>
+          experiments.runCommand(
+            AvailableCommands.EXPERIMENT_BRANCH,
+            dvcRoot,
+            name,
+            input
+          )
+      )
+  )
+
+  internalCommands.registerExternalCliCommand(
+    RegisteredCliCommands.EXPERIMENT_VIEW_SHARE,
+    ({ dvcRoot, id }: ExperimentDetails) =>
+      experiments.getExpNameAndInputThenRun(
+        Title.ENTER_BRANCH_NAME,
+        dvcRoot,
+        id,
+        async (name: string, input: string) => {
+          await experiments.runCommand(
+            AvailableCommands.EXPERIMENT_BRANCH,
+            dvcRoot,
+            name,
+            input
+          )
+          await experiments.runCommand(
+            AvailableCommands.EXPERIMENT_APPLY,
+            dvcRoot,
+            name
+          )
+          await experiments.runCommand(AvailableCommands.PUSH, dvcRoot)
+          return Toast.showOutput(gitPushBranch(dvcRoot, input))
```

Comment:
[C] This makes me think that we should be showing all of the git commands that we run inside of our DVC output channel.

I will follow up and give Git its own reader/executor.